src/celeritas/em/distribution/UrbanMscScatter.hh

```diff
@@ -201,29 +202,34 @@ CELER_FUNCTION auto UrbanMscScatter::operator()(Engine& rng) -> MscInteraction
     real_type phi
         = UniformRealDistribution<real_type>(0, 2 * constants::pi)(rng);
 
+    MscInteraction result;
+    result.action = MscInteraction::Action::scattered;
+    {
+        // This should only be needed to silence compiler warning
+        result.displacement = {0, 0, 0};
+    }
+
     // Calculate displacement
-    Real3 displacement;
     if (input_.is_displaced && tau_ >= params_.tau_small)
     {
         // Sample displacement and adjust
-        displacement    = this->sample_displacement(rng, phi);
-        real_type rmax2 = (true_path_ - input_.geom_path)
-                          * (true_path_ + input_.geom_path);
-        real_type scaling
-            = this->calc_displacement_scaling(displacement, rmax2);
-        for (auto i : range(3))
+        real_type length = this->calc_displacement_length(
+            (true_path_ - input_.geom_path) * (true_path_ + input_.geom_path));
```

Comment:
I thought that we were going to drop the input_ data member and use individual components (i.e, replace input_.geom_path by geom_path_).

Reply:
Yep, that's in your MR: this one was just to add the action result.